src/Microsoft.Crank.Agent/Startup.cs

```diff
@@ -3835,24 +3837,28 @@ private static string SeekCompatibleDesktopRuntime(string dotnetHome, string tar
             return desktopVersion;
         }
 
-        private static async Task<string> ResolveAspNetCoreVersion(string aspNetCoreVersion, string currentAspNetCoreVersion, string targetFramework)
+        private static async Task<string> ResolveAspNetCoreVersion(string aspNetCoreVersion, string targetFramework)
         {
+            (_, _, var currentAspNetCoreVersion, _) = await GetCurrentVersions(targetFramework);
             var versionPrefix = targetFramework.Substring(targetFramework.Length - 3);
 
             // Define which ASP.NET Core packages version to use
 
             switch (aspNetCoreVersion.ToLowerInvariant())
             {
                 case "current":
-                    aspNetCoreVersion = currentAspNetCoreVersion;
+                    aspNetCoreVersion = string.IsNullOrEmpty(currentAspNetCoreVersion)
+                        ? await ResolveAspNetCoreVersion("Latest", targetFramework)
+                        : currentAspNetCoreVersion;
+
                     Log.Info($"ASP.NET: {aspNetCoreVersion} (Current)");
                     break;
                 case "latest":
                     // aspnet runtime service releases are not published on feeds
                     switch (versionPrefix)
                     {
                         case "9.0":
-                            var productsInfo = JObject.Parse(await DownloadContentAsync(_latestProductVersions90Url));
+                            var productsInfo = JObject.Parse(await DownloadContentAsync(_latestProductVersions90Url, cache: _latestProductVersions90Cache));
```

Comment:
It looks like `_latestProductVersions90Cache` is the actual memory cache here. I would rename both the static and the parameter to include timeout.

```suggestion
                            var productsInfo = JObject.Parse(await DownloadContentAsync(_latestProductVersions90Url, cacheTimeout: _latestProductVersions90CacheTimeout));
```

Same goes for `_aspnetSdkVersionCache` -> `_aspnetSdkVersionCacheTimeout`.